### ISSUE TITLE
Fix Request for AvoidDiscardingHttpClientConfig - false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#425]
In the following case it is only supplied in the setter and not in the constructor:
```
import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
import org.springframework.http.client.ClientHttpRequestFactory;

public class Issue {
    private ClientHttpRequestFactory getClientHttpRequestFactory(HttpClientConfiguration httpClientConfiguration) {
        HttpComponentsClientHttpRequestFactory httpComponentsClientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
        CloseableHttpClient httpClient = createHttpClient(httpClientConfiguration);
        httpComponentsClientHttpRequestFactory.setHttpClient(httpClient);  // <--- false positive
        return httpComponentsClientHttpRequestFactory;
    }
}
```
